saleor/graphql/discount/tests/queries/test_sales_sorting.py

```diff
@@ -1,84 +1,77 @@
-                discount_value=7,
-                sale=sales[0],
-                channel=channel_PLN,
-                currency=channel_PLN.currency_code,
+            PromotionRule(
+                promotion=promotions[0],
+                reward_value=7,
+                reward_value_type=RewardValueType.PERCENTAGE,
             ),
-            SaleChannelListing(
-                discount_value=7,
-                sale=sales[1],
-                channel=channel_USD,
-                currency=channel_USD.currency_code,
+            PromotionRule(
+                promotion=promotions[1],
+                reward_value=7,
+                reward_value_type=RewardValueType.FIXED,
             ),
-            SaleChannelListing(
-                discount_value=1,
-                sale=sales[1],
-                channel=channel_PLN,
-                currency=channel_PLN.currency_code,
+            PromotionRule(
+                promotion=promotions[1],
+                reward_value=1,
+                reward_value_type=RewardValueType.FIXED,
             ),
-            SaleChannelListing(
-                discount_value=5,
-                sale=sales[2],
-                channel=channel_USD,
-                currency=channel_USD.currency_code,
+            PromotionRule(
+                promotion=promotions[2],
+                reward_value=5,
+                reward_value_type=RewardValueType.PERCENTAGE,
             ),
-            SaleChannelListing(
-                discount_value=2,
-                sale=sales[3],
-                channel=channel_PLN,
-                currency=channel_PLN.currency_code,
+            PromotionRule(
+                promotion=promotions[3],
+                reward_value=2,
+                reward_value_type=RewardValueType.FIXED,
             ),
-            SaleChannelListing(
-                discount_value=2,
-                sale=sales[4],
-                channel=channel_USD,
-                currency=channel_USD.currency_code,
+            PromotionRule(
+                promotion=promotions[4],
+                reward_value=2,
+                reward_value_type=RewardValueType.FIXED,
             ),
-            SaleChannelListing(
-                discount_value=5,
-                sale=sales[4],
-                channel=channel_PLN,
-                currency=channel_PLN.currency_code,
+            PromotionRule(
+                promotion=promotions[4],
+                reward_value=5,
+                reward_value_type=RewardValueType.FIXED,
             ),
         ]
     )
-
-    convert_sales_to_promotions()
-    promotions = Promotion.objects.order_by("created_at").all()
+    rules[0].channels.add(channel_USD)
+    rules[1].channels.add(channel_PLN)
+    rules[2].channels.add(channel_USD)
+    rules[3].channels.add(channel_PLN)
+    rules[4].channels.add(channel_USD)
+    rules[5].channels.add(channel_PLN)
+    rules[6].channels.add(channel_USD)
+    rules[7].channels.add(channel_PLN)
```

Comment:
Each `add` is performing DB query with `insert`. So IMO we should add rules to channels instead of adding channels to rules to reduce number of DB queries